pycrostates/segmentation/transitions.py

```diff
@@ -156,3 +156,55 @@ def _compute_expected_transition_matrix(
         T_expected = T_expected * 100
 
     return T_expected
+
+
+def _check_labels_n_clusters(
+    labels: NDArray[int],
+    n_clusters: int,
+) -> None:
+    """Checker for labels and n_clusters."""
+    _check_type(labels, (np.ndarray,), "labels")
+    _check_type(n_clusters, ("int",), "n_clusters")
+
+    # only accept array of integers to simplify maintenance
+    if labels.dtype != "int":
+        raise ValueError(
+            "The argument 'labels' must contain the labels of each timepoint "
+            "encoded as consecutive positive integers (0-indexed). Make sure "
+            f"you are providing an integer array. '{labels.dtype}' is "
+            "invalid."
+        )
+    # check for negative integers except -1
+    if np.any(labels < -1):
+        raise ValueError(
+            "The argument 'labels' must contain the labels of each timepoint "
+            "encoded as consecutive positive integers (0-indexed) with '-1' "
+            "representing unlabelled segments. Negative integers except -1 "
+            "are invalid."
+        )
+    # check for NaN
+    if np.any(np.isnan(labels)):
+        raise ValueError(
+            "The argument 'labels' must contain the labels of each timepoint "
+            "encoded as consecutive positive integers (0-indexed) with '-1' "
+            "representing unlabelled segments. 'np.nan' are invalid."
+        )
+    # check for consecutiveness and 0-index
+    states = sorted([elt for elt in np.unique(labels) if 0 <= elt])
+    if (
+        states[0] != 0
+        or len(states) != np.arange(states[0], states[-1] + 1).size
+    ):
+        raise ValueError(
+            "The argument 'labels' must contain the labels of each timepoint "
+            "encoded as consecutive positive integers (0-indexed), with the "
+            f"first state encoded as '0'. '{states}' are invalid."
+        )
+    # check if the states match the number of clusters
+    if len(states) != n_clusters:
+        raise ValueError(
+            "The argument 'labels' must contain the labels of each timepoint "
+            "encoded as consecutive positive integers (0-indexed), with the "
+            f"first state encoded as '0'. '{states}' are invalid with "
+            f"'n_clusters' set to '{n_clusters}'."
+        )
```

Comment:
If would rather handle the case where one state never occur in the segmentation ( i.e. `labels = [1,2,3,4]` and` n_clusters=[5]`). Something like:

```python
if not np.isin(states, np.arange(n_clusters)):
    raise ...
```

Reply:
I started with more constrained 'labels' and 'n_clusters', but indeed that should work nicely. I can't think of a breaking edge case. I changed the logic and the tests accordingly.